idea/src/org/jetbrains/kotlin/idea/intentions/ConvertBinaryExpressionWithDemorgansLawIntention.kt

```diff
@@ -31,50 +31,57 @@ class ConvertBinaryExpressionWithDemorgansLawIntention : SelfTargetingOffsetInde
 
     override fun applyTo(element: KtBinaryExpression, editor: Editor?) = applyTo(element)
 
-    fun applyTo(element: KtBinaryExpression) {
-        val expr = element.parentsWithSelf.takeWhile { it is KtBinaryExpression }.last() as KtBinaryExpression
-
-        val operatorText = when (expr.operationToken) {
-            KtTokens.ANDAND -> KtTokens.OROR.value
-            KtTokens.OROR -> KtTokens.ANDAND.value
-            else -> throw IllegalArgumentException()
+    companion object {
+        fun convertIfPossible(element: KtBinaryExpression) {
+            val expr = element.parentsWithSelf.takeWhile { it is KtBinaryExpression }.last() as KtBinaryExpression
```

Comment:
I think it'd be better to extract `element.parentsWithSelf.takeWhile { it is KtBinaryExpression }.last() as KtBinaryExpression` to a function

Reply:
955365e